api/tests/mocha/services/users.spec.js

```diff
@@ -674,6 +675,574 @@ describe('Users service', () => {
+            password: 'short',
+          },
+          {
+            username: 'x',
+            place: 'x',
+            contact: { parent: 'x' },
+            type: 'national-manager',
+            password: 'password',
+          },
+        ]);
+        chai.assert.fail('Should have thrown');
+      } catch (error) {
+        // short password
+        chai.expect(error.failingIndexes[0].passwordError.message.message).to.equal(
+          'The password must be at least 8 characters long.',
+        );
+        chai.expect(error.failingIndexes[0].passwordError.message.translationKey).to.equal(
+          'password.length.minimum',
+        );
+        chai.expect(error.failingIndexes[0].passwordError.message.translationParams).to.have.property('minimum');
+        chai.expect(error.failingIndexes[0].index).to.equal(0);
+
+        // weak password
+        chai.expect(error.failingIndexes[1].passwordError.message.message).to.equal(
+          'The password is too easy to guess. Include a range of types of characters to increase the score.',
+        );
+        chai.expect(error.failingIndexes[1].passwordError.message.translationKey).to.equal('password.weak');
+        chai.expect(error.failingIndexes[1].index).to.equal(1);
+
+        chai.expect(error.code).to.equal(400);
+      }
+    });
+
+    it('returns error if one of the users has a missing phone number', async () => {
+      const tokenLoginConfig = { translation_key: 'sms', enabled: true };
+      sinon.stub(config, 'get')
+        .withArgs('token_login').returns(tokenLoginConfig)
+        .withArgs('app_url').returns('url');
+      sinon.stub(auth, 'isOffline').returns(false);
+
+      try {
+        await service.createUsers([
+          {
+            username: 'sally',
+            roles: ['a', 'b'],
+            phone: '123',
+            token_login: true,
+          },
+          {
+            username: 'sally',
+            roles: ['a', 'b'],
+            token_login: true,
+          },
+        ]);
+        chai.assert.fail('Should have thrown');
+      } catch (error) {
+        chai.expect(error.code).to.equal(400);
+        chai.expect(error.failingIndexes[0].fields).to.deep.equal(['phone']);
+        chai.expect(error.failingIndexes[0].index).to.equal(1);
+      }
+    });
+
+    it('returns error if one of the users has an invalid phone number', async () => {
+      const tokenLoginConfig = { translation_key: 'sms', enabled: true };
+      sinon.stub(config, 'get')
+        .withArgs('token_login').returns(tokenLoginConfig)
+        .withArgs('app_url').returns('url');
+      sinon.stub(auth, 'isOffline').returns(false);
+
+      try {
+        await service.createUsers([
+          {
+            username: 'sally',
+            roles: ['a', 'b'],
+            phone: '+40 755 69-69-69',
+            token_login: true,
+          },
+          {
+            username: 'sally',
+            roles: ['a', 'b'],
+            phone: '123',
```

Comment:
Please use a valid phone number here.

Reply:
This test here asserts the phone number is invalid for the user at index 1. Is there a correct phone number format that is considered invalid by `google-libphonenumber`?

Reply:
Oh, so sorry, I was mislead by the [0]. It's great!